seismicpro/src/gather/gather.py

```diff
@@ -1028,6 +1029,41 @@ def crop(self, origins, crop_shape, n_crops=1, stride=None, pad_mode='constant',
         origins = make_origins(origins, self.shape, crop_shape, n_crops, stride)
         return CroppedGather(self, origins, crop_shape, pad_mode, **kwargs)
 
+    @batch_method(target="for")
+    def apply_agc(self, factor=1, window_size=250, mode='rms'):
+        """ TODO """
+        # Cast window from ms to samples
+        window_size_samples = window_size // int(self.sample_rate)
```

Comment:
```suggestion
        window_size_samples = int(window_size / self.sample_rate)
```
Current version may return wrong size in case of float sample rate